src/main/kotlin/org/rust/lang/core/psi/impl/mixin/RsImplItemImplMixin.kt

```diff
@@ -31,3 +30,19 @@ abstract class RsImplItemImplMixin : RsStubbedElementImpl<RsImplItemStub>, RsImp
         return PresentationData(typeReference?.text ?: "Impl", null, RsIcons.IMPL, null)
     }
 }
+
+/**
+ * @return pair of two lists: (mandatory trait members, optional trait members)
+ */
+fun RsImplItem.toImplementOverride(): Pair<List<RsNamedElement>, List<RsNamedElement>> {
+    val trait = traitRef?.trait ?: error("No trait ref")
+    val traitMembers = trait.functionList as List<RsAbstractable> + trait.typeAliasList + trait.constantList 
```

Comment:
`traitMembers = trait.children.filterIsInstance<RsAbstractable>()` will cover all possible elements without naming them explicitly and without creating several temporary lists. The same with `members`.